04-materials/06-working-on-your-own.md

```diff
@@ -17,50 +17,241 @@ We will be using the [official template](https://github.com/jupyterlab/extension
+is high enough and the confetti container is properly positioned.
+```
+
+To customize:
+```
+Make the confetti more colorful and add a sound effect when it triggers.
+```
+
+**Verified Setup**:
+- ✅ macOS 15.7, Claude Code (Claude Sonnet 4.5)
+- ✅ `conda` environment: Python 3.13, Node.js 22, JupyterLab
+- 📹 [Watch the demo](https://www.loom.com/share/2afabea0184045fa868271f9ab0ca083)
+
+:::{tip}
+This is a perfect first project - it's self-contained, purely visual, and you'll immediately see if it works. It teaches you about JupyterLab's status bar API without backend complexity.
+:::
+
+#### 🎨 Custom theme extension
+
+**Complexity**: Beginner (Frontend only)
+**Time**: 20-30 minutes
+**What you'll learn**: Theme customization, CSS styling, JupyterLab theming system
+
+**The Prompt**:
+```
+Create a theme based on [your favorite movie/show/game/aesthetic]
+```
+
+**Example**:
+```
+Create a theme based on Netflix show KPop Demon Hunters
+```
 
+**Expected results**:
+- A new theme appears in Settings → Theme menu
+- Theme includes custom colors matching your chosen aesthetic
+- JupyterLab interface reflects the theme's visual style
+- Theme can be toggled on/off
 
-### For Server Extensions:
+**What to watch for**:
 
-- **Jupyter Server Handlers**: [Documentation](https://jupyter-server.readthedocs.io/en/latest/developers/extensions.html)
-- **Tornado Request Handlers**: [Tornado Docs](https://www.tornadoweb.org/en/stable/web.html)
+To add a background image to the main panel:
+```
+Can we use this image as a background for main panel?
+[paste your image URL]
+```
 
+Example:
+```
+Can we use this image as a background for main panel?
+https://www.billboard.com/wp-content/uploads/2025/07/kpop-demon-hunters-billboard-1800.jpg?w=942&h=628&crop=1
+```
 
-## 🐛 Beginner-Friendly GitHub
+To refine the styling:
+```
+The background image is too bright - make it more subtle with reduced opacity
+and add a dark overlay so text is readable.
+```
 
+To adjust colors:
+```
+Update the sidebar and toolbar colors to match the theme's color palette.
+Use [specific colors] for accent elements.
+```
+
+**Tips for finding images**:
+- Search "[theme name] wallpaper 4k" for high-quality backgrounds
+- Look for images with good contrast for text readability
+- Consider color palettes - extract 3-5 main colors from your chosen image (AI can help you with it!)
+- Free image sources: Unsplash, Pexels, Wallhaven
+
+:::{tip}
+This is a perfect creative project! You get immediate visual feedback, can personalize your JupyterLab environment, and learn how JupyterLab's theming system works. Plus, you'll have a custom theme you actually want to use daily.
+
+Themes are also great conversation starters - share your theme with other workshop participants!
+:::
+
+#### 📊 CPU monitor widget
+
+**Complexity**: Intermediate (Frontend + Server)
+**Time**: 30-40 minutes
+**What you'll learn**: REST API integration, backend data processing, graceful error handling
```

Comment:
```suggestion
**What you'll work on**: REST API integration, backend data processing, error handling
```

If we are moving away from learning objectives, maybe we can phrase this slightly differently?